src/Native/String.js

```diff
@@ -192,11 +192,16 @@ function indexes(sub, str)
 	var subLen = sub.length;
 	var i = 0;
 	var is = [];
-	while ((i = str.indexOf(sub, i)) > -1)
+	
+	if (subLen > 0)
 	{
-		is.push(i);
-		i = i + subLen;
+		while ((i = str.indexOf(sub, i)) > -1)
+		{
+			is.push(i);
+			i = i + subLen;
+		}	
 	}
+	
```

Comment:
Better, but it can be optimized a bit more :) If you move the `if` above allocating `i` and `is` you can just write something like this:

```javascript
if (subLen > 0)
{
    return _elm_lang$core$Native_List.Nil;
}
```

Which is cool in two ways. First, it means `i` and `is` do not even get allocated. Second, and more importantly, it uses the [early return](http://blog.timoxley.com/post/47041269194/avoid-else-return-early) style, which I try to use in any C-like language I use. As a reader, it means I can be sure, nothing past this point is relevant to this code / some invariant is definitely true past this point.